### ISSUE TITLE
Docker Desktop 4.25.0 WSL Integration is broken on Ubuntu 22.04 LTS

### ISSUE THREAD

**Issue by edyu** [docker/for-win#13764]
### Description

I have the latest WSL2 (2.0.6.0) and the latest Ubuntu 22.04.3 LTS (which is my default distro) and I also have an Alpine Linux WSL distro as well. I have 2 machines with same setups.
I installed Docker Desktop on both machines and enabled WSL2 Integration. I have checked `Enable Integration with my default WSL distro` and also `Enable integration with additional distros` with both Ubuntu and Alpine enabled.
In both cases, I was able to run `docker info` on Alpine.
However for Ubuntu 22.04, I have different problems:
1. For one of the machine, when I run docker I see the following error:
```
 I  ~ ❱ docker

The command 'docker' could not be found in this WSL 2 distro.
We recommend to activate the WSL integration in Docker Desktop settings.

For details about using Docker Desktop with WSL 2, visit:

https://docs.docker.com/go/wsl2/
```
2. For the other machine, when I run docker I see the following error:
```
Server:
ERROR: Cannot connect to the Docker daemon at unix:///var/run/docker.sock. Is the docker daemon running?
errors pretty printing info
```

### Reproduce

1. Install WSL2 and Ubuntu 22.04.3
2. Install Docker Desktop 4.25.0
3. Run `docker` in the Ubuntu WSL

### Expected behavior

docker should connect to Docker Desktop on Windows and no errors

### docker version

```bash
Client:
 Cloud integration: v1.0.35+desktop.5
 Version:           24.0.6
 API version:       1.43
 Go version:        go1.20.7
 Git commit:        ed223bc
 Built:             Mon Sep  4 12:32:48 2023
 OS/Arch:           windows/amd64
 Context:           default

Server: Docker Desktop 4.25.0 (126437)
 Engine:
  Version:          24.0.6
  API version:      1.43 (minimum version 1.12)
  Go version:       go1.20.7
  Git commit:       1a79695
  Built:            Mon Sep  4 12:32:16 2023
  OS/Arch:          linux/amd64
  Experimental:     false
 containerd:
  Version:          1.6.22
  GitCommit:        8165feabfdfe38c65b599c4993d227328c231fca
 runc:
  Version:          1.1.8
  GitCommit:        v1.1.8-0-g82f18fe
 docker-init:
  Version:          0.19.0
  GitCommit:        de40ad0
```

### docker info

```bash
Client:
 Version:    24.0.6
 Context:    default
 Debug Mode: false
 Plugins:
  buildx: Docker Buildx (Docker Inc.)
    Version:  v0.11.2-desktop.5
    Path:     C:\Program Files\Docker\cli-plugins\docker-buildx.exe
  compose: Docker Compose (Docker Inc.)
    Version:  v2.23.0-desktop.1
    Path:     C:\Program Files\Docker\cli-plugins\docker-compose.exe
  dev: Docker Dev Environments (Docker Inc.)
    Version:  v0.1.0
    Path:     C:\Program Files\Docker\cli-plugins\docker-dev.exe
  extension: Manages Docker extensions (Docker Inc.)
    Version:  v0.2.20
    Path:     C:\Program Files\Docker\cli-plugins\docker-extension.exe
  init: Creates Docker-related starter files for your project (Docker Inc.)
    Version:  v0.1.0-beta.9
    Path:     C:\Program Files\Docker\cli-plugins\docker-init.exe
  sbom: View the packaged-based Software Bill Of Materials (SBOM) for an image (Anchore Inc.)
    Version:  0.6.0
    Path:     C:\Program Files\Docker\cli-plugins\docker-sbom.exe
  scan: Docker Scan (Docker Inc.)
    Version:  v0.26.0
    Path:     C:\Program Files\Docker\cli-plugins\docker-scan.exe
  scout: Docker Scout (Docker Inc.)
    Version:  v1.0.9
    Path:     C:\Program Files\Docker\cli-plugins\docker-scout.exe

Server:
 Containers: 0
  Running: 0
  Paused: 0
  Stopped: 0
 Images: 0
 Server Version: 24.0.6
 Storage Driver: overlay2
  Backing Filesystem: extfs
  Supports d_type: true
  Using metacopy: false
  Native Overlay Diff: true
  userxattr: false
 Logging Driver: json-file
 Cgroup Driver: cgroupfs
 Cgroup Version: 2
 Plugins:
  Volume: local
  Network: bridge host ipvlan macvlan null overlay
  Log: awslogs fluentd gcplogs gelf journald json-file local logentries splunk syslog
 Swarm: inactive
 Runtimes: io.containerd.runc.v2 runc
 Default Runtime: runc
 Init Binary: docker-init
 containerd version: 8165feabfdfe38c65b599c4993d227328c231fca
 runc version: v1.1.8-0-g82f18fe
 init version: de40ad0
 Security Options:
  seccomp
   Profile: unconfined
  cgroupns
 Kernel Version: 5.15.133.1-microsoft-standard-WSL2
 Operating System: Docker Desktop
 OSType: linux
 Architecture: x86_64
 CPUs: 20
 Total Memory: 15.47GiB
 Name: area51
 ID: 97f04ee8-1fb8-41a8-b7d0-cc08d9374df2
 Docker Root Dir: /var/lib/docker
 Debug Mode: false
 HTTP Proxy: http.docker.internal:3128
 HTTPS Proxy: http.docker.internal:3128
 No Proxy: hubproxy.docker.internal
 Experimental: false
 Insecure Registries:
  hubproxy.docker.internal:5555
  127.0.0.0/8
 Live Restore Enabled: false

WARNING: daemon is not using the default seccomp profile
```

### Diagnostics ID

D15F1CD0-0E1E-4BE1-AC26-A8242136C3D6/20231027050450

**Comment by priisholm** [comment]
+1, Same problem here

**Comment by 7-zete-7** [comment]
Same problem on WLinux WSL2

**Comment by 7-zete-7** [comment]
Downgrading to 4.24.2 helps.

**Comment by avogt1701** [comment]
Had the same issue. Docker reboot now suddenly has fixed the issue.

Docker Version 4.25.0
WSL-Version: 1.2.5.0

**Comment by bsousaa** [comment]
@edyu @priisholm @7-zete-7 can you check if reboot fixes the issue?

**Comment by 7-zete-7** [comment]
@bsousaa, reboot does not fixes the issue. And factory reset does not fixes.

**Comment by bsousaa** [comment]
Thanks @7-zete-7 can you share a diagnostics id?

**Comment by 7-zete-7** [comment]
@bsousaa, гnfortunately no. I downgrade to 4.24.2 after several reboots and several factory resets because none of this helped and I needed to quickly restore functionality.

**Comment by hhromic** [comment]
I also started experiencing random "docker not found" errors inside of WSL after upgrading to v4.25.0. Especially after not using Docker after some time (no containers running).

After some investigation, I suspect that the resource saver function is removing the symlinks inside of WSL when entering into resource saving mode. When the "docker not found" errors start appearing, I simply open the Docker Desktop UI window (which brings Docker back from resource saver mode) and then after a short while the "docker" command inside of WSL starts working again, without restarting anything.

Hope this helps finding the culprit.

**Comment by edyu** [comment]
> @edyu @priisholm @7-zete-7 can you check if reboot fixes the issue? 

Nope. Unfortunately. I rebooted multiple times already.

**Comment by edyu** [comment]
> I also started experiencing random "docker not found" errors inside of WSL after upgrading to v4.25.0. Especially after not using Docker after some time (no containers running).
> 
> After some investigation, I suspect that the resource saver function is removing the symlinks inside of WSL when entering into resource saving mode. When the "docker not found" errors start appearing, I simply open the Docker Desktop UI window (which brings Docker back from resource saver mode) and then after a short while the "docker" command inside of WSL starts working again, without restarting anything.
> 
> Hope this helps finding the culprit.

I'll try this today. I'm surprised Ubuntu has so many problems as it's the "default" WSL2 setup.

**Comment by edyu** [comment]
> @edyu @priisholm @7-zete-7 can you check if reboot fixes the issue? 

One additional information. I've enabled systemd and automount on my Ubuntu WSL2 using `/etc/wsl.conf`

**Comment by edyu** [comment]
> I also started experiencing random "docker not found" errors inside of WSL after upgrading to v4.25.0. Especially after not using Docker after some time (no containers running).
> 
> After some investigation, I suspect that the resource saver function is removing the symlinks inside of WSL when entering into resource saving mode. When the "docker not found" errors start appearing, I simply open the Docker Desktop UI window (which brings Docker back from resource saver mode) and then after a short while the "docker" command inside of WSL starts working again, without restarting anything.
> 
> Hope this helps finding the culprit.

It doesn't work for me. I've even disabled resource saver.

**Comment by hhromic** [comment]
> It doesn't work for me. I've even disabled resource saver.

Ah then likely a different issue for you than for me. Or perhaps related but triggering differently.

I have been paying attention today and I can fully confirm now that, at least for me, every time Docker Desktop enters into resource saving mode, the "docker" command is removed from my WSL environment. When I go to the Docker Desktop UI and awaken Docker to get out of resource saving mode, the "docker" command is restored in WSL without restarting anything.

This did not happen in previous versions of Docker Desktop (<v4.25.0) where the "docker" command was always available and triggered the resource saver mode to be deactivate when active.

EDIT: I can also confirm now that the same happens in another Windows 10 machine I have with Docker Desktop v4.25.0.

**Comment by edyu** [comment]
I can verify that @7-zete-7 is correct! Docker Desktop 4.24.2 works but 4.25.0 doesn't.

**Comment by priisholm** [comment]
> @edyu @priisholm @7-zete-7 can you check if reboot fixes the issue? 

It didn't. Tried multiple tweaks in configuration and reboots, but unfortunately none of them fixed the problem, so I had to downgrade to get work done.

**Comment by edyu** [comment]
> +1, Same problem here

I suggest downgrading Docker Desktop 4.25.0 to 4.24.2

**Comment by priisholm** [comment]
> > +1, Same problem here
> 
> I suggest downgrading Docker Desktop 4.25.0 to 4.24.2 

Yes, I did that and it works fine now, thank you.

Btw, I originally installed DD, upgraded to v. 4.25, uninstalled and reinstalled v. 4.24.2 using Chocolatey w.o. any hassles, except for time wasted.

**Comment by papparapa** [comment]
I had the same problem and solved it by downgrading to 4.24.2.
Thanks.

**Comment by andhisan** [comment]
Seems like symlinks broken after upgrading to 4.25. After I manually linked these, I cannot use plugins including buildx or compose.
Anyway, I had no choice but to downgrade.

**Comment by parberge** [comment]
Some more weird behavior I have found:

I can start **individual** containers via docker desktop UI.
Trying to start a compose service via docker desktop UI yields:

```
Cannot start Docker Compose application. Reason: compose [start] exit status 1. Cannot connect to the Docker daemon at unix:///var/run/docker.sock. Is the docker daemon running?
```

And also as many mentioned before `docker` inside my Ubuntu 22.04 wsl doesn't work at all anymore.

**Comment by agarzon** [comment]
same problem, but with **debian** WSL too.

**Comment by oorestisime** [comment]
Same issue with debian. Tried to fix the symlinks but then compose is not working.

**Comment by agarzon** [comment]
The only thing that worked, was downgrading to the previous version 4.24.2 😐

**Comment by lorenrh** [comment]
Hello all!

we would like to investigate further, could some of you share your diagnostics so we can investigate further? 

Thanks!

**Comment by parberge** [comment]
@lorenrh Not sure about the process, but ran `DIAGNOSTICS` and uploaded it via the docker desktop app.
This is the ID I got after uploading: `4889EACE-9574-4CE1-93C6-9E8A8E9BBE0F/20231031145937`

**Comment by lorenrh** [comment]
Yes @parberge  thanks!

For info, there's a [troubleshoot guide](https://docs.docker.com/desktop/troubleshoot/overview/#diagnose)?

If it's not possible from the UI, you also get the diagnostics from the terminal, see [Diagnose from the terminal](https://docs.docker.com/desktop/troubleshoot/overview/#diagnose-from-the-terminal).

**Comment by parberge** [comment]
> Yes @parberge thanks!
> 
> For info, you [troubleshoot guide](https://docs.docker.com/desktop/troubleshoot/overview/#diagnose)?
> 
> If it's not possible from the UI, you also get the diagnostics from the terminal, see [Diagnose from the terminal](https://docs.docker.com/desktop/troubleshoot/overview/#diagnose-from-the-terminal).

Yeah exactly, everything via docker desktop app:

<img width="1895" alt="image" src="https://github.com/docker/for-win/assets/5441497/43c12f5f-472e-428d-8b87-ec50a1121ef9">

**Comment by agarzon** [comment]
Mine 4309F943-4147-4265-B168-08B36A293419/20231031150707

all [troubleshoot guide](https://docs.docker.com/desktop/troubleshoot/overview/#diagnose) was done including terminal

**Comment by chelnak** [comment]
Thanks everyone!

**Comment by tmuka** [comment]
me too. downgrading

**Comment by chelnak** [comment]
Hey @tmuka, any chance you could share a diagnostic with us before you downgrade? 🙏

**Comment by chelnak** [comment]
Hey everyone,

If you have time, please could you test [this build](https://desktop-stage.docker.com/win/main/amd64/127055/Docker%20Desktop%20Installer.exe), and let us know if the issue still persists?

**Comment by parberge** [comment]
> Hey everyone,
> 
> If you have time, please could you test [this build](https://desktop-stage.docker.com/win/main/amd64/127055/Docker%20Desktop%20Installer.exe), and let us know if the issue still persists?

@chelnak Seems to be working (haven't tested much, but at least the error I got before isn't happening anymore)

**Comment by chelnak** [comment]
This is really good news! Thanks for testing it out.

**Comment by agarzon** [comment]
> Hey everyone,
> 
> If you have time, please could you test [this build](https://desktop-stage.docker.com/win/main/amd64/127055/Docker%20Desktop%20Installer.exe), and let us know if the issue still persists?

Did not worked for me.

Diagnostic 4309F943-4147-4265-B168-08B36A293419/20231102140803

**Comment by tmuka** [comment]
thanks for the fast turnaround. 
installing and uninstalling then reinstalling the test build Docker Desktop 4.25.0 (127055) didn't work for me. Same result on two different Windows 11 machines. Still erroring inside WSL `docker ps` 
`Cannot connect to the Docker daemon at unix:///var/run/docker.sock. Is the docker daemon running?`

Diagnostic ID: 24EF6714-5042-429A-B293-B8C6803DFD46/20231102141206

**Comment by parberge** [comment]
> > Hey everyone,
> > If you have time, please could you test [this build](https://desktop-stage.docker.com/win/main/amd64/127055/Docker%20Desktop%20Installer.exe), and let us know if the issue still persists?
> 
> @chelnak Seems to be working (haven't tested much, but at least the error I got before isn't happening anymore)

Yeah sorry for the false positive. Don't know why, but it stopped working. Maybe it was after a system restart. Not sure.

**Comment by chelnak** [comment]
@tmuka @agarzon @parberge Thank you!

Back to the drawing board then.

@parberge can you send a diagnostic when you get some free time? I'm interested that something seemed to work then stopped.

Also, if you haven't already, can you all try a factory reset please?

**Comment by tmuka** [comment]
> @tmuka @agarzon @parberge Thank you!
> 
> Back to the drawing board then.
> 
> @parberge can you send a diagnostic when you get some free time? I'm interested that something seemed to work then stopped.
> 
> Also, if you haven't already, can you all try a factory reset please?

Sure thing.  I did try a factory reset, and deleting all data.  and toggling WSL integration for default, and additional containers.

**Comment by chelnak** [comment]
Appreciate the info thanks!

**Comment by chm10** [comment]
This issue is expected in Ubuntu, but with other distributions that use WSL with Docker Desktop too, I am having the same problem with Debian.

> Installing and uninstalling and then reinstalling the test build Docker Desktop 4.25.0 (127055) didn't work for me. 
> Cannot connect to the Docker daemon at unix:///var/run/docker.sock. Is the docker daemon running?

I am having an issue sending the diagnostic. 

`Diagnostics were not uploaded. Reason: Error: {"message":"[2023-11-03T03:58:58.613699400Z][com.docker.diagnose.exe.settingsloader][I] settings get: {\"acceptCanaryUpdates\":false,\"activeOrganizationName\":\"\",\"allowExperimentalFeatures\":true,\"analyticsEnabled\":true,\"autoDownloadUpdates\":true,\"autoPauseTimedActivitySeconds\":30,\"autoPauseTimeoutSeconds\":300,\"autoStart\":false,\"backupData\":false,\"containerTerminal\":\"integrated\",\"cpus\":8,\"credentialHelper\":\"docker-credential-wincred.exe\",\"customWslDistroDir\":\"\",\"dataFolder\":\"C:\\\\ProgramData\\\\DockerDesktop\\\\vm-data\",\"deprecatedCgroupv1\":false,\"disableHardwareAcceleration\":false,\"disableUpdate\":false,\"diskFlush\":\"\",\"diskSizeMiB\":65536,\"diskStats\":\"\",\"diskTRIM\":true,\"displayRestartDialog\":true,\"displaySwitchWinLinContainers\":true,\"displayed18362Deprecation\":false,\"displayedElectronPopup\":[],\"displayedOnboarding\":true,\"dockerAppLaunchPath\":\"\",\"dockerBinInstallPath\":\"system\",\"enableDefaultDockerSocket\":true,\"enableIntegrationWithDefaultWslDistro\":true,\"enableIntegrityCheck\":true,\"enableSegmentDebug\":false,\"enableWasmShims\":false,\"enhancedContainerIsolation\":false,\"exposeDockerAPIOnTCP2375\":false,\"extensionsEnabled\":true,\"extensionsPrivateMarketplace\":false,\"extensionsPrivateMarketplaceAdminContactURL\":\"\",\"filesharingDirectories\":[],\"integratedWslDistros\":[\"WLinux\"],\"kernelForUDP\":false,\"kubernetesEnabled\":false,\"kubernetesInitialInstallPerformed\":false,\"lastLoginDate\":0,\"latestBannerKey\":\"\",\"licenseTermsVersion\":2,\"lifecycleTimeoutSeconds\":600,\"memoryMiB\":2048,\"networkType\":\"gvisor\",\"noWindowsContainers\":false,\"onlyMarketplaceExtensions\":false,\"openUIOnStartupDisabled\":false,\"overrideProxyExclude\":\"\",\"overrideProxyHttp\":\"\",\"overrideProxyHttps\":\"\",\"overrideWindowsDockerdPort\":\"float64\",\"proxyHttpMode\":\"s...m\",\"requireVmnetd\":false,\"runWinServiceInWslMode\":false,\"sbomIndexing\":true,\"settingsVersion\":35,\"showAnnouncementNotifications\":true,\"showBuildsViewFeature\":false,\"showExtensionsSystemContainers\":false,\"showGeneralNotifications\":true,\"showInstallScreen\":true,\"showKubernetesSystemContainers\":false,\"showSurveyNotifications\":true,\"skipUpdateToWSLPrompt\":true,\"skipWSLMountPerfWarning\":false,\"socksProxyPort\":0,\"swapMiB\":1024,\"synchronizedDirectories\":[],\"themeSource\":\"system\",\"updateAvailableTime\":0,\"updateHostsFile\":false,\"updateInstallTime\":0,\"useBackgroundIndexing\":true,\"useContainerdSnapshotter\":false,\"useCredentialHelper\":true,\"useGrpcfuse\":true,\"useNightlyBuildUpdates\":false,\"useResourceSaver\":true,\"useVirtualizationFramework\":false,\"useVirtualizationFrameworkRosetta\":false,\"useVirtualizationFrameworkVirtioFS\":false,\"useVpnkit\":true,\"useWindowsContainers\":false,\"vpnKitAllowedBindAddresses\":\"\",\"vpnKitMTU\":0,\"vpnKitMaxConnections\":0,\"vpnKitMaxPortIdleTime\":\"float64\",\"vpnKitTransparentProxy\":true,\"vpnkitCIDR\":\"1...4\",\"wslEnableGrpcfuse\":false,\"wslEngineEnabled\":true,\"wslInstallMode\":\"installLatestWsl\",\"wslUpdateRequired\":false}\n[2023-11-03T03:58:58.613699400Z][com.docker.diagnose.exe.httpproxy][I] SetSettings: manual override disabled\n[2023-11-03T03:58:58.618042000Z][com.docker.diagnose.exe.httpproxy][I] SetSystemProxies: using per-URL HTTP proxy settings\n[2023-11-03T03:58:58.618042000Z][com.docker.diagnose.exe.httpproxy][I] SetAdminSettings: admin settings disabled\n[2023-11-03T03:58:58.618255600Z][com.docker.diagnose.exe.public][I] using HTTP proxy inside com.docker.backend\n: exit status 1"}`

**Comment by mimikun** [comment]
I have same issue.
In my case, it happened after updating from Win11 22H2 to 23H2.
Diagnostics ID is `F9CB060B-EBC6-4EF1-9A23-77AA7634E244/20231104062250`

Is it better to downgrade to v4.24.2?

**Comment by chelnak** [comment]
Thanks for reporting the issue 🙂

Quick question. Do you have anything custom (containers/service) configured to run in the background or at startup? Also, do you have anything executing in your bashrc/profile that is non standard?

**Comment by mimikun** [comment]
My shell configuration file is below.

https://github.com/mimikun/dotfiles/blob/master/OneDrive/Documents/PowerShell/Microsoft.PowerShell_profile.ps1.tmpl
https://github.com/mimikun/dotfiles/blob/master/dot_config/fish/config.fish.tmpl

Also, set `restart: always` in docker-compose.yml [mimikun/mastodon-markov-bot](https://github.com/mimikun/mastodon-markov-bot/blob/main/docker-compose.yml) and `selenium/standalone-chrome` were running.

The following apps start at startup:
- Cloudflare WARP
- Non-stick Mouse

**Comment by chelnak** [comment]
Hey! Thank you. This is really useful.

**Comment by zoltan-magyar** [comment]
I have a similar problem. Docker Desktop 4.25.0. The WSL2 distro integrations work, as they symlink the docker executables in the distros, but for some reason, the actual socket is always missing. I have 3 distros, all WSL2, all worked previously. Arch, Ubuntu and openSUSE. They all have the same problem.

I did dive into the `com.docker.wsl-distro-proxy.exe` logs and it's quite clear that it doesn't seem to attempt symlinking the socket:

```
[2023-11-03T11:13:19.049827100Z][com.docker.wsl-distro-proxy.exe][I] start wsl integration for distro Arch
[2023-11-03T11:13:19.054020600Z][com.docker.wsl-distro-proxy.exe.linuxworkspaces][I] wsl.exe -d Arch -e sh -c mkdir -p ~/.docker/run
[2023-11-03T11:13:19.188213300Z][com.docker.wsl-distro-proxy.exe.linuxworkspaces][I] wsl.exe -d Arch -e sh -c cat ~/.docker/config.json
[2023-11-03T11:13:19.327365200Z][com.docker.wsl-distro-proxy.exe.linuxworkspaces][I] wsl.exe -d Arch -e sh -c cat - > ~/.docker/config.json
[2023-11-03T11:13:19.485908900Z][com.docker.wsl-distro-proxy.exe.linuxworkspaces][I] wsl.exe -d Arch -e sh -c mkdir -p ~/.docker/scan
[2023-11-03T11:13:19.642777000Z][com.docker.wsl-distro-proxy.exe.linuxworkspaces][I] wsl.exe -d Arch -e sh -c cat ~/.docker/scan/config.json
[2023-11-03T11:13:19.797726200Z][com.docker.wsl-distro-proxy.exe.linuxworkspaces][I] wsl.exe -d Arch -e sh -c cat - > ~/.docker/scan/config.json
[2023-11-03T11:13:19.937246800Z][com.docker.wsl-distro-proxy.exe.linuxworkspaces][I] wsl.exe -d Arch -e whoami
[2023-11-03T11:13:20.061885700Z][com.docker.wsl-distro-proxy.exe.linuxworkspaces][I] wsl.exe -d Arch -u root -e sh -c ( addgroup docker || true ) && addgroup "zoltanmagyar" docker
[2023-11-03T11:13:20.187279600Z][com.docker.wsl-distro-proxy.exe.linuxworkspaces][I] wsl.exe -d Arch -u root -e sh -c ( groupadd docker || true ) && usermod -aG docker "zoltanmagyar"
[2023-11-03T11:13:20.311499200Z][com.docker.wsl-distro-proxy.exe.linuxworkspaces][I] wsl.exe -d Arch -e cat /etc/wsl.conf
[2023-11-03T11:13:20.435471300Z][com.docker.wsl-distro-proxy.exe.linuxworkspaces][I] wsl.exe -d Arch -e cat /etc/wsl.conf
[2023-11-03T11:13:20.560449100Z][com.docker.wsl-distro-proxy.exe.linuxworkspaces][I] wsl.exe -d Arch -e /mnt/wsl/docker-desktop/docker-desktop-user-distro sync-context-store /mnt/c/Users/zoltanmagyar/.docker/contexts
[2023-11-03T11:13:20.684724200Z][com.docker.wsl-distro-proxy.exe.linuxworkspaces][I] wsl.exe -d Arch -e cat /etc/wsl.conf
[2023-11-03T11:13:20.808098300Z][com.docker.wsl-distro-proxy.exe.linuxworkspaces][I] wsl.exe -d Arch -e cat /etc/wsl.conf
[2023-11-03T11:13:20.919386500Z][com.docker.wsl-distro-proxy.exe.linuxworkspaces][I] wsl.exe -d Arch -e /mnt/wsl/docker-desktop/docker-desktop-user-distro sync-home-subdir /mnt/c/Users/zoltanmagyar/.aws
[2023-11-03T11:13:21.042712900Z][com.docker.wsl-distro-proxy.exe.linuxworkspaces][I] wsl.exe -d Arch -e cat /etc/wsl.conf
[2023-11-03T11:13:21.165481100Z][com.docker.wsl-distro-proxy.exe.linuxworkspaces][I] wsl.exe -d Arch -e cat /etc/wsl.conf
[2023-11-03T11:13:21.288797700Z][com.docker.wsl-distro-proxy.exe.linuxworkspaces][I] wsl.exe -d Arch -e /mnt/wsl/docker-desktop/docker-desktop-user-distro sync-home-subdir /mnt/c/Users/zoltanmagyar/.azure
[2023-11-03T11:13:21.413985000Z][com.docker.wsl-distro-proxy.exe.linuxworkspaces][I] wsl.exe -d Arch -e cat /etc/wsl.conf
[2023-11-03T11:13:21.539519400Z][com.docker.wsl-distro-proxy.exe.linuxworkspaces][I] wsl.exe -d Arch -e cat /etc/wsl.conf
[2023-11-03T11:13:21.664778800Z][com.docker.wsl-distro-proxy.exe.linuxworkspaces][I] wsl.exe -d Arch -e /mnt/wsl/docker-desktop/docker-desktop-user-distro sync-file /mnt/c/Users/zoltanmagyar/.docker/features.json ~/.docker/features.json
-------------------------------------------------------------------------------->8
-------------------------------------------------------------------------------->8
[2023-11-04T18:10:35.079702200Z][com.docker.wsl-distro-proxy.exe][I] kill leaked integration processes for distro Arch
[2023-11-04T18:10:35.084608900Z][com.docker.wsl-distro-proxy.exe.linuxworkspaces][I] wsl.exe -d Arch -u root -e sh -c ps aux
``` 

I found logs where it works, from previous versions (Worth noting that I don't know what other distro's had integration turned on, here I think Ubuntu was off, but Arch always was, as it's my default distro):
```
[2023-09-25T04:47:27.681696500Z][com.docker.wsl-distro-proxy.exe][I] start wsl integration for distro Arch
[2023-09-25T04:47:27.682245000Z][com.docker.wsl-distro-proxy.exe.linuxworkspaces][I] wsl.exe -d Arch -e sh -c mkdir -p ~/.docker/run
[2023-09-25T04:47:27.923932800[2023-09-25T04:47:27.923932800Z][com.docker.wsl-distro-proxy.exe.linuxworkspaces][I] wsl.exe -d openSUSE-T[2023-09-25T04:47:28.129631900Z][com.docker.wsl-distro-proxy.exe.linuxworkspaces][I] wsl.exe -d Arch -e sh -c cat - > ~/.docker/config.json
[2023-09-25T04:47:28.315758300Z][com.docker.wsl-distro-proxy[2023-09-25T04:47:28.315758300Z][com.docker.wsl-distro-proxy.exe.linuxwork[2023-09-25T04:47:28.517692900Z][com.docker.wsl-distro-proxy.exe.linuxworks[2023-09-25T04:47:28.533364300Z][com.docker.wsl-distro-proxy.exe.l[2023-09-25T04:47:28.736959200Z][com.docker.wsl-distro-proxy.exe.linuxworkspaces][I] wsl.e[2023-09-25T04:47:28.752337400Z][com.docker.wsl-distro-[2023-09-25T04:47:28.939468900Z][com.docker.wsl-distro-proxy.exe.linuxworkspaces][I] wsl.exe -d Arch -e w[2023-[2023-09-25T04:47:29.218815900Z][com.docker.wsl-distro-proxy.exe.linuxworkspaces][I] wsl.exe -d Arch -u root -e sh -c ( addgroup docker || true ) && addgroup "zoltanmagyar" docker
[2023-09-25T04:47:29.419566700Z][com.docker.wsl-distro-proxy.exe.linuxworkspaces][I] wsl.exe -d Arch -u root -e sh -c ( groupadd docker || true ) && usermod -aG docker "zoltanmagyar"
[2023-09-25T04:47:29.734374400Z][com.docker.wsl-distro-proxy.exe.linuxworkspaces][I] wsl.exe -d Arch -e cat /etc/wsl.conf
[2023-09-25T04:47:30.150657900Z][com.docker.wsl-distro-proxy.exe.linuxworkspaces][I] wsl.exe -d Arch -e cat /etc/wsl.conf
[2023-09-25T04:47:30.448192700Z][com.docker.wsl-distro-proxy.exe.linuxworkspaces][I] wsl.exe -d Arch -e /mnt/wsl/docker-desktop/docker-desktop-user-distro sync-context-store /mnt/c/Users/zoltanmagyar/.docker/contexts
[2023-09-25T04:47:30.681916200Z][com.docker.wsl-distro-proxy.exe.linuxworkspaces][I] wsl.exe -d Arch -e cat /etc/wsl.conf
[2023-09-25T04:47:30.867932500Z][com.docker.wsl-distro-proxy.exe.linuxworkspaces][I] wsl.exe -d Arch -e cat /etc/wsl.conf
[2023-09-25T04:47:31.054108400Z][com.docker.wsl-distro-proxy.exe.linuxworkspaces][I] wsl.exe -d Arch -e /mnt/wsl/docker-desktop/docker-desktop-user-distro sync-home-subdir /mnt/c/Users/zoltanmagyar/.aws
[2023-09-25T04:47:31.254929500Z][com.docker.wsl-distro-proxy.exe.linuxworkspaces][I] wsl.exe -d Arch -e cat /etc/wsl.conf
[2023-09-25T04:47:31.456582300Z][com.docker.wsl-distro-proxy.exe.linuxworkspaces][I] wsl.exe -d Arch -e cat /etc/wsl.conf
[2023-09-25T04:47:31.656216100Z][com.docker.wsl-distro-proxy.exe.linuxworkspaces][I] wsl.exe -d Arch -e /mnt/wsl/docker-desktop/docker-desktop-user-distro sync-home-subdir /mnt/c/Users/zoltanmagyar/.azure
[2023-09-25T04:47:31.843695100Z][com.docker.wsl-distro-proxy.exe.linuxworkspaces][I] wsl.exe -d Arch -e cat /etc/wsl.conf
[2023-09-25T04:47:32.015228000Z][com.docker.wsl-distro-proxy.exe.linuxworkspaces][I] wsl.exe -d Arch -e cat /etc/wsl.conf
[2023-09-25T04:47:32.172443300Z][com.docker.wsl-distro-proxy.exe.linuxworkspaces][I] wsl.exe -d Arch -e /mnt/wsl/docker-desktop/docker-desktop-user-distro sync-file /mnt/c/Users/zoltanmagyar/.docker/features.json ~/.docker/features.json
[2023-09-25T04:47:32.342724600Z][com.docker.wsl-distro-proxy.exe.linuxworkspaces][I] wsl.exe -d Arch -e sh -c echo $HOME
[2023-09-25T04:47:32.497777200Z][com.docker.wsl-distro-proxy.exe.linuxworkspaces][I] Killing CLI API server running at /home/zoltanmagyar/.docker/run/docker-cli-api.sock
[2023-09-25T04:47:32.497777200Z][com.docker.wsl-distro-proxy.exe.linuxworkspaces][I] wsl.exe -d Arch -e sh -c ps -x
[2023-09-25T04:47:32.651810000Z][com.docker.wsl-distro-proxy.exe.linuxworkspaces][I] wsl.exe -d Arch -e sh -c kill 399
[2023-09-25T04:47:32.807631200Z][com.docker.wsl-distro-proxy.exe.linuxworkspaces][I] wsl.exe -d Arch -e ls /home/zoltanmagyar/.docker/run/docker-cli-api.sock
[2023-09-25T04:47:32.978654500Z][com.docker.wsl-distro-proxy.exe][I] Starting WSL Distro proxy 2
[2023-09-25T04:47:32.978654500Z][com.docker.wsl-distro-proxy.exe.linuxworkspaces][I] wsl.exe -d Arch -e cat /etc/wsl.conf
[2023-09-25T04:47:33.134263300Z][com.docker.wsl-distro-proxy.exe.linuxworkspaces][I] wsl.exe -d Arch -u root -e /mnt/wsl/docker-desktop/docker-desktop-user-distro proxy --distro-name Arch --docker-desktop-root /mnt/wsl/docker-desktop C:\Program Files\Docker\Docker\resources
[2023-09-25T04:47:36.305071500Z][com.docker.wsl-distro-proxy.exe.linuxworkspaces][I] Starting distro cli api server on /home/zoltanmagyar/.docker/run/docker-cli-api.sock
[2023-09-25T04:47:36.305071500Z][com.docker.wsl-distro-proxy.exe.linuxworkspaces][I] wsl.exe -d Arch -e docker serve --address unix:///home/zoltanmagyar/.docker/run/docker-cli-api.sock
[2023-09-25T04:49:02.892917200Z][com.docker.wsl-distro-proxy.exe][I] docker-desktop-user-distro proxy has exited
[2023-09-25T04:49:02.892917200Z][com.docker.wsl-distro-proxy.exe][I] Killing WSL CLI API server
[2023-09-25T04:49:02.892917200Z][com.docker.wsl-distro-proxy.exe.linuxworkspaces][I] Killing CLI API server running at /home/zoltanmagyar/.docker/run/docker-cli-api.sock
[2023-09-25T04:49:02.892917200Z][com.docker.wsl-distro-proxy.exe.linuxw[2023-09-25T04:49:03.727295800Z][com.docker.wsl-distro-proxy.exe][I] docker-desktop-user-distro proxy has exited
[2023-09-25T04:49:03.727295800Z][com.docker.wsl-distro-proxy.exe][I] Killing WSL CLI API server
[2023-09-25T04:49:03.727295800Z][com.docker.wsl-distro-proxy.exe.linuxworkspaces][I] Killing CLI API server running at /home/zoltanmagyar/.docker/run/docker-cli-api.sock
[2023-09-25T04:49:03.727295800Z][com.docker.wsl-distro-proxy.exe.linuxworkspaces][I] wsl.exe -d openSUSE-Tumbleweed -e sh -c ps -x
[2023-09-25T04:49:03.849561300Z][com.docker.wsl-distro-proxy.exe][I] Killing WSL Distro proxy
[2023-09-25T04:49:03.849561300Z][com.docker.wsl-distro-proxy.exe.linuxworkspaces][I] wsl.exe -d openSUSE-Tumbleweed -u root -e sh -c kill $(cat /run/docker-desktop-proxy.pid)
[2023-09-25T04:49:03.974068800Z][com.docker.wsl-distro-proxy.exe][W] Distro Proxy Kill error: exit status 2
[2023-09-25T04:49:03.974210400Z][com.docker.wsl-distro-proxy.exe][I] exit wsl integration for distro openSUSE-Tumbleweed
:49:02.816373000Z][com.docker.wsl-distro-proxy.exe.linuxworkspaces][I] wsl.exe -d Arch -u root -e sh -c kill $(cat /run/docker-desktop-proxy.pid)
[2023-09-25T04:49:02.939662800Z][com.docker.wsl-distro-proxy.exe.linuxworkspaces][I] wsl.exe -d Arch -e cat /etc/wsl.conf
[2023-09-25T04:49:03.061853300Z][com.docker.wsl-distro-proxy.exe.linuxworkspaces][I] wsl.exe -d Arch -u root -e /mnt/wsl/docker-desktop/docker-desktop-user-distro --docker-desktop-root /mnt/wsl/docker-desktop --distro-name Arch cleanup-cli-tools
-------------------------------------------------------------------------------->8
-------------------------------------------------------------------------------->8
[2023-09-25T04:49:03.218865000Z][com.docker.wsl-distro-proxy.exe][I] kill integration processes for distro openSUSE-Tumbleweed
[2023-09-25T04:49:03.218865000Z][com.docker.wsl-distro-proxy.exe.linuxworkspaces][I] wsl.exe -d openSUSE-Tumbleweed -e sh -c echo $HOME
[2023-09-25T04:49:03.339495000Z][com.docker.wsl-distro-proxy.exe][I] Killing WSL CLI API server
[2023-09-25T04:49:03.339495000Z][com.docker.wsl-distro-proxy.exe.linuxworkspaces][I] Killing CLI API server running at /home/zoltanmagyar/.docker/run/docker-cli-api.sock
[2023-09-25T04:49:03.339495000Z][com.docker.wsl-distro-proxy.exe.linuxworkspaces][I] wsl.exe -d openSUSE-Tumbleweed -e sh -c ps -x
[2023-09-25T04:49:03.463989900Z][com.docker.wsl-distro-proxy.exe.linuxworkspaces][I] wsl.exe -d openSUSE-Tumbleweed -e sh -c kill 633
[2023-09-25T04:49:03.635452800Z][com.docker.wsl-distro-proxy.exe][I] Killing WSL Distro proxy
[2023-09-25T04:49:03.635452800Z][com.docker.wsl-distro-proxy.exe.linuxworkspaces][I] wsl.exe -d openSUSE-Tumbleweed -u root -e sh -c kill $(cat /run/docker-desktop-proxy.pid)
[2023-09-25T04:49:03.758441100Z][com.docker.wsl-distro-proxy.exe.linuxworkspaces][I] wsl.exe -d openSUSE-Tumbleweed -e cat /etc/wsl.conf
[2023-09-25T04:49:03.880806400Z][com.docker.wsl-distro-proxy.exe.linuxworkspaces][I] wsl.exe -d openSUSE-Tumbleweed -u root -e /mnt/wsl/docker-desktop/docker-desktop-user-distro --docker-desktop-root /mnt/wsl/docker-desktop --distro-name openSUSE-Tumbleweed cleanup-cli-tools
-------------------------------------------------------------------------------->8
-------------------------------------------------------------------------------->8
[2023-10-10T00:56:14.922696300Z][com.docker.wsl-distro-proxy.exe][I] kill leaked integration processes for distro Arch
[2023-10-10T00:56:14.924861800Z][com.docker.wsl-distro-proxy.exe.linuxworkspaces][I] wsl.exe -d Arch -u root -e sh -c ps aux
-------------------------------------------------------------------------------->8
-------------------------------------------------------------------------------->8
[2023-10-10T00:56:15.308897600Z][com.docker.wsl-distro-proxy.exe][I] kill leaked integration processes for distro openSUSE-Tumbleweed
[2023-10-10T00:56:15.309421800Z][com.docker.wsl-distro-proxy.exe.linuxworkspaces][I] wsl.exe -d openSUSE-Tumbleweed -u root -e sh -c ps aux
```

**Comment by chelnak** [comment]
@zoltan-magyar thank you for the detailed report!

In these distros do you have any containers/services set to run on boot or anything custom being executed in your login profile (bash/fish/zsh/etc)?

**Comment by zoltan-magyar** [comment]
@chelnak 

No containers or services, I do use systemd ever since it's been a feature on WSL2 on all distros.
I also use fish with all of these distros, with an identical set of plugins:

- jorgebucaran/fisher
- jorgebucaran/nvm.fish
- blackjid/plugin-kubectl

And the following `config.fish` (which basically justh initializes the starship prompt):

```
starship init fish | source

if status is-interactive
    # Commands to run in interactive sessions can go here
end
```

**Comment by edyu** [comment]
> I have same issue. In my case, it happened after updating from Win11 22H2 to 23H2. Diagnostics ID is `F9CB060B-EBC6-4EF1-9A23-77AA7634E244/20231104062250`
> 
> Is it better to downgrade to v4.24.2?

yes

**Comment by edyu** [comment]
> Thanks for reporting the issue 🙂
> 
> Quick question. Do you have anything custom (containers/service) configured to run in the background or at startup? Also, do you have anything executing in your bashrc/profile that is non standard?

nothing special at all.  nothing special as i don't even use bash. I do use `fish` on my ubuntu but nothing special. The containers are all alpine which uses ash by default.

**Comment by 58bits** [comment]
Same problem here - burned up two hours this morning trying to fix this.

**Comment by zoltan-magyar** [comment]
> I have a similar problem. Docker Desktop 4.25.0. The WSL2 distro integrations work, as they symlink the docker executables in the distros, but for some reason, the actual socket is always missing. I have 3 distros, all WSL2, all worked previously. Arch, Ubuntu and openSUSE. They all have the same problem.
> 
> I did dive into the `com.docker.wsl-distro-proxy.exe` logs and it's quite clear that it doesn't seem to attempt symlinking the socket:
> 
> ```
> [2023-11-03T11:13:19.049827100Z][com.docker.wsl-distro-proxy.exe][I] start wsl integration for distro Arch
> [2023-11-03T11:13:19.054020600Z][com.docker.wsl-distro-proxy.exe.linuxworkspaces][I] wsl.exe -d Arch -e sh -c mkdir -p ~/.docker/run
> [2023-11-03T11:13:19.188213300Z][com.docker.wsl-distro-proxy.exe.linuxworkspaces][I] wsl.exe -d Arch -e sh -c cat ~/.docker/config.json
> [2023-11-03T11:13:19.327365200Z][com.docker.wsl-distro-proxy.exe.linuxworkspaces][I] wsl.exe -d Arch -e sh -c cat - > ~/.docker/config.json
> [2023-11-03T11:13:19.485908900Z][com.docker.wsl-distro-proxy.exe.linuxworkspaces][I] wsl.exe -d Arch -e sh -c mkdir -p ~/.docker/scan
> [2023-11-03T11:13:19.642777000Z][com.docker.wsl-distro-proxy.exe.linuxworkspaces][I] wsl.exe -d Arch -e sh -c cat ~/.docker/scan/config.json
> [2023-11-03T11:13:19.797726200Z][com.docker.wsl-distro-proxy.exe.linuxworkspaces][I] wsl.exe -d Arch -e sh -c cat - > ~/.docker/scan/config.json
> [2023-11-03T11:13:19.937246800Z][com.docker.wsl-distro-proxy.exe.linuxworkspaces][I] wsl.exe -d Arch -e whoami
> [2023-11-03T11:13:20.061885700Z][com.docker.wsl-distro-proxy.exe.linuxworkspaces][I] wsl.exe -d Arch -u root -e sh -c ( addgroup docker || true ) && addgroup "zoltanmagyar" docker
> [2023-11-03T11:13:20.187279600Z][com.docker.wsl-distro-proxy.exe.linuxworkspaces][I] wsl.exe -d Arch -u root -e sh -c ( groupadd docker || true ) && usermod -aG docker "zoltanmagyar"
> [2023-11-03T11:13:20.311499200Z][com.docker.wsl-distro-proxy.exe.linuxworkspaces][I] wsl.exe -d Arch -e cat /etc/wsl.conf
> [2023-11-03T11:13:20.435471300Z][com.docker.wsl-distro-proxy.exe.linuxworkspaces][I] wsl.exe -d Arch -e cat /etc/wsl.conf
> [2023-11-03T11:13:20.560449100Z][com.docker.wsl-distro-proxy.exe.linuxworkspaces][I] wsl.exe -d Arch -e /mnt/wsl/docker-desktop/docker-desktop-user-distro sync-context-store /mnt/c/Users/zoltanmagyar/.docker/contexts
> [2023-11-03T11:13:20.684724200Z][com.docker.wsl-distro-proxy.exe.linuxworkspaces][I] wsl.exe -d Arch -e cat /etc/wsl.conf
> [2023-11-03T11:13:20.808098300Z][com.docker.wsl-distro-proxy.exe.linuxworkspaces][I] wsl.exe -d Arch -e cat /etc/wsl.conf
> [2023-11-03T11:13:20.919386500Z][com.docker.wsl-distro-proxy.exe.linuxworkspaces][I] wsl.exe -d Arch -e /mnt/wsl/docker-desktop/docker-desktop-user-distro sync-home-subdir /mnt/c/Users/zoltanmagyar/.aws
> [2023-11-03T11:13:21.042712900Z][com.docker.wsl-distro-proxy.exe.linuxworkspaces][I] wsl.exe -d Arch -e cat /etc/wsl.conf
> [2023-11-03T11:13:21.165481100Z][com.docker.wsl-distro-proxy.exe.linuxworkspaces][I] wsl.exe -d Arch -e cat /etc/wsl.conf
> [2023-11-03T11:13:21.288797700Z][com.docker.wsl-distro-proxy.exe.linuxworkspaces][I] wsl.exe -d Arch -e /mnt/wsl/docker-desktop/docker-desktop-user-distro sync-home-subdir /mnt/c/Users/zoltanmagyar/.azure
> [2023-11-03T11:13:21.413985000Z][com.docker.wsl-distro-proxy.exe.linuxworkspaces][I] wsl.exe -d Arch -e cat /etc/wsl.conf
> [2023-11-03T11:13:21.539519400Z][com.docker.wsl-distro-proxy.exe.linuxworkspaces][I] wsl.exe -d Arch -e cat /etc/wsl.conf
> [2023-11-03T11:13:21.664778800Z][com.docker.wsl-distro-proxy.exe.linuxworkspaces][I] wsl.exe -d Arch -e /mnt/wsl/docker-desktop/docker-desktop-user-distro sync-file /mnt/c/Users/zoltanmagyar/.docker/features.json ~/.docker/features.json
> -------------------------------------------------------------------------------->8
> -------------------------------------------------------------------------------->8
> [2023-11-04T18:10:35.079702200Z][com.docker.wsl-distro-proxy.exe][I] kill leaked integration processes for distro Arch
> [2023-11-04T18:10:35.084608900Z][com.docker.wsl-distro-proxy.exe.linuxworkspaces][I] wsl.exe -d Arch -u root -e sh -c ps aux
> ```
> 
> I found logs where it works, from previous versions (Worth noting that I don't know what other distro's had integration turned on, here I think Ubuntu was off, but Arch always was, as it's my default distro):
> 
> ```
> [2023-09-25T04:47:27.681696500Z][com.docker.wsl-distro-proxy.exe][I] start wsl integration for distro Arch
> [2023-09-25T04:47:27.682245000Z][com.docker.wsl-distro-proxy.exe.linuxworkspaces][I] wsl.exe -d Arch -e sh -c mkdir -p ~/.docker/run
> [2023-09-25T04:47:27.923932800[2023-09-25T04:47:27.923932800Z][com.docker.wsl-distro-proxy.exe.linuxworkspaces][I] wsl.exe -d openSUSE-T[2023-09-25T04:47:28.129631900Z][com.docker.wsl-distro-proxy.exe.linuxworkspaces][I] wsl.exe -d Arch -e sh -c cat - > ~/.docker/config.json
> [2023-09-25T04:47:28.315758300Z][com.docker.wsl-distro-proxy[2023-09-25T04:47:28.315758300Z][com.docker.wsl-distro-proxy.exe.linuxwork[2023-09-25T04:47:28.517692900Z][com.docker.wsl-distro-proxy.exe.linuxworks[2023-09-25T04:47:28.533364300Z][com.docker.wsl-distro-proxy.exe.l[2023-09-25T04:47:28.736959200Z][com.docker.wsl-distro-proxy.exe.linuxworkspaces][I] wsl.e[2023-09-25T04:47:28.752337400Z][com.docker.wsl-distro-[2023-09-25T04:47:28.939468900Z][com.docker.wsl-distro-proxy.exe.linuxworkspaces][I] wsl.exe -d Arch -e w[2023-[2023-09-25T04:47:29.218815900Z][com.docker.wsl-distro-proxy.exe.linuxworkspaces][I] wsl.exe -d Arch -u root -e sh -c ( addgroup docker || true ) && addgroup "zoltanmagyar" docker
> [2023-09-25T04:47:29.419566700Z][com.docker.wsl-distro-proxy.exe.linuxworkspaces][I] wsl.exe -d Arch -u root -e sh -c ( groupadd docker || true ) && usermod -aG docker "zoltanmagyar"
> [2023-09-25T04:47:29.734374400Z][com.docker.wsl-distro-proxy.exe.linuxworkspaces][I] wsl.exe -d Arch -e cat /etc/wsl.conf
> [2023-09-25T04:47:30.150657900Z][com.docker.wsl-distro-proxy.exe.linuxworkspaces][I] wsl.exe -d Arch -e cat /etc/wsl.conf
> [2023-09-25T04:47:30.448192700Z][com.docker.wsl-distro-proxy.exe.linuxworkspaces][I] wsl.exe -d Arch -e /mnt/wsl/docker-desktop/docker-desktop-user-distro sync-context-store /mnt/c/Users/zoltanmagyar/.docker/contexts
> [2023-09-25T04:47:30.681916200Z][com.docker.wsl-distro-proxy.exe.linuxworkspaces][I] wsl.exe -d Arch -e cat /etc/wsl.conf
> [2023-09-25T04:47:30.867932500Z][com.docker.wsl-distro-proxy.exe.linuxworkspaces][I] wsl.exe -d Arch -e cat /etc/wsl.conf
> [2023-09-25T04:47:31.054108400Z][com.docker.wsl-distro-proxy.exe.linuxworkspaces][I] wsl.exe -d Arch -e /mnt/wsl/docker-desktop/docker-desktop-user-distro sync-home-subdir /mnt/c/Users/zoltanmagyar/.aws
> [2023-09-25T04:47:31.254929500Z][com.docker.wsl-distro-proxy.exe.linuxworkspaces][I] wsl.exe -d Arch -e cat /etc/wsl.conf
> [2023-09-25T04:47:31.456582300Z][com.docker.wsl-distro-proxy.exe.linuxworkspaces][I] wsl.exe -d Arch -e cat /etc/wsl.conf
> [2023-09-25T04:47:31.656216100Z][com.docker.wsl-distro-proxy.exe.linuxworkspaces][I] wsl.exe -d Arch -e /mnt/wsl/docker-desktop/docker-desktop-user-distro sync-home-subdir /mnt/c/Users/zoltanmagyar/.azure
> [2023-09-25T04:47:31.843695100Z][com.docker.wsl-distro-proxy.exe.linuxworkspaces][I] wsl.exe -d Arch -e cat /etc/wsl.conf
> [2023-09-25T04:47:32.015228000Z][com.docker.wsl-distro-proxy.exe.linuxworkspaces][I] wsl.exe -d Arch -e cat /etc/wsl.conf
> [2023-09-25T04:47:32.172443300Z][com.docker.wsl-distro-proxy.exe.linuxworkspaces][I] wsl.exe -d Arch -e /mnt/wsl/docker-desktop/docker-desktop-user-distro sync-file /mnt/c/Users/zoltanmagyar/.docker/features.json ~/.docker/features.json
> [2023-09-25T04:47:32.342724600Z][com.docker.wsl-distro-proxy.exe.linuxworkspaces][I] wsl.exe -d Arch -e sh -c echo $HOME
> [2023-09-25T04:47:32.497777200Z][com.docker.wsl-distro-proxy.exe.linuxworkspaces][I] Killing CLI API server running at /home/zoltanmagyar/.docker/run/docker-cli-api.sock
> [2023-09-25T04:47:32.497777200Z][com.docker.wsl-distro-proxy.exe.linuxworkspaces][I] wsl.exe -d Arch -e sh -c ps -x
> [2023-09-25T04:47:32.651810000Z][com.docker.wsl-distro-proxy.exe.linuxworkspaces][I] wsl.exe -d Arch -e sh -c kill 399
> [2023-09-25T04:47:32.807631200Z][com.docker.wsl-distro-proxy.exe.linuxworkspaces][I] wsl.exe -d Arch -e ls /home/zoltanmagyar/.docker/run/docker-cli-api.sock
> [2023-09-25T04:47:32.978654500Z][com.docker.wsl-distro-proxy.exe][I] Starting WSL Distro proxy 2
> [2023-09-25T04:47:32.978654500Z][com.docker.wsl-distro-proxy.exe.linuxworkspaces][I] wsl.exe -d Arch -e cat /etc/wsl.conf
> [2023-09-25T04:47:33.134263300Z][com.docker.wsl-distro-proxy.exe.linuxworkspaces][I] wsl.exe -d Arch -u root -e /mnt/wsl/docker-desktop/docker-desktop-user-distro proxy --distro-name Arch --docker-desktop-root /mnt/wsl/docker-desktop C:\Program Files\Docker\Docker\resources
> [2023-09-25T04:47:36.305071500Z][com.docker.wsl-distro-proxy.exe.linuxworkspaces][I] Starting distro cli api server on /home/zoltanmagyar/.docker/run/docker-cli-api.sock
> [2023-09-25T04:47:36.305071500Z][com.docker.wsl-distro-proxy.exe.linuxworkspaces][I] wsl.exe -d Arch -e docker serve --address unix:///home/zoltanmagyar/.docker/run/docker-cli-api.sock
> [2023-09-25T04:49:02.892917200Z][com.docker.wsl-distro-proxy.exe][I] docker-desktop-user-distro proxy has exited
> [2023-09-25T04:49:02.892917200Z][com.docker.wsl-distro-proxy.exe][I] Killing WSL CLI API server
> [2023-09-25T04:49:02.892917200Z][com.docker.wsl-distro-proxy.exe.linuxworkspaces][I] Killing CLI API server running at /home/zoltanmagyar/.docker/run/docker-cli-api.sock
> [2023-09-25T04:49:02.892917200Z][com.docker.wsl-distro-proxy.exe.linuxw[2023-09-25T04:49:03.727295800Z][com.docker.wsl-distro-proxy.exe][I] docker-desktop-user-distro proxy has exited
> [2023-09-25T04:49:03.727295800Z][com.docker.wsl-distro-proxy.exe][I] Killing WSL CLI API server
> [2023-09-25T04:49:03.727295800Z][com.docker.wsl-distro-proxy.exe.linuxworkspaces][I] Killing CLI API server running at /home/zoltanmagyar/.docker/run/docker-cli-api.sock
> [2023-09-25T04:49:03.727295800Z][com.docker.wsl-distro-proxy.exe.linuxworkspaces][I] wsl.exe -d openSUSE-Tumbleweed -e sh -c ps -x
> [2023-09-25T04:49:03.849561300Z][com.docker.wsl-distro-proxy.exe][I] Killing WSL Distro proxy
> [2023-09-25T04:49:03.849561300Z][com.docker.wsl-distro-proxy.exe.linuxworkspaces][I] wsl.exe -d openSUSE-Tumbleweed -u root -e sh -c kill $(cat /run/docker-desktop-proxy.pid)
> [2023-09-25T04:49:03.974068800Z][com.docker.wsl-distro-proxy.exe][W] Distro Proxy Kill error: exit status 2
> [2023-09-25T04:49:03.974210400Z][com.docker.wsl-distro-proxy.exe][I] exit wsl integration for distro openSUSE-Tumbleweed
> :49:02.816373000Z][com.docker.wsl-distro-proxy.exe.linuxworkspaces][I] wsl.exe -d Arch -u root -e sh -c kill $(cat /run/docker-desktop-proxy.pid)
> [2023-09-25T04:49:02.939662800Z][com.docker.wsl-distro-proxy.exe.linuxworkspaces][I] wsl.exe -d Arch -e cat /etc/wsl.conf
> [2023-09-25T04:49:03.061853300Z][com.docker.wsl-distro-proxy.exe.linuxworkspaces][I] wsl.exe -d Arch -u root -e /mnt/wsl/docker-desktop/docker-desktop-user-distro --docker-desktop-root /mnt/wsl/docker-desktop --distro-name Arch cleanup-cli-tools
> -------------------------------------------------------------------------------->8
> -------------------------------------------------------------------------------->8
> [2023-09-25T04:49:03.218865000Z][com.docker.wsl-distro-proxy.exe][I] kill integration processes for distro openSUSE-Tumbleweed
> [2023-09-25T04:49:03.218865000Z][com.docker.wsl-distro-proxy.exe.linuxworkspaces][I] wsl.exe -d openSUSE-Tumbleweed -e sh -c echo $HOME
> [2023-09-25T04:49:03.339495000Z][com.docker.wsl-distro-proxy.exe][I] Killing WSL CLI API server
> [2023-09-25T04:49:03.339495000Z][com.docker.wsl-distro-proxy.exe.linuxworkspaces][I] Killing CLI API server running at /home/zoltanmagyar/.docker/run/docker-cli-api.sock
> [2023-09-25T04:49:03.339495000Z][com.docker.wsl-distro-proxy.exe.linuxworkspaces][I] wsl.exe -d openSUSE-Tumbleweed -e sh -c ps -x
> [2023-09-25T04:49:03.463989900Z][com.docker.wsl-distro-proxy.exe.linuxworkspaces][I] wsl.exe -d openSUSE-Tumbleweed -e sh -c kill 633
> [2023-09-25T04:49:03.635452800Z][com.docker.wsl-distro-proxy.exe][I] Killing WSL Distro proxy
> [2023-09-25T04:49:03.635452800Z][com.docker.wsl-distro-proxy.exe.linuxworkspaces][I] wsl.exe -d openSUSE-Tumbleweed -u root -e sh -c kill $(cat /run/docker-desktop-proxy.pid)
> [2023-09-25T04:49:03.758441100Z][com.docker.wsl-distro-proxy.exe.linuxworkspaces][I] wsl.exe -d openSUSE-Tumbleweed -e cat /etc/wsl.conf
> [2023-09-25T04:49:03.880806400Z][com.docker.wsl-distro-proxy.exe.linuxworkspaces][I] wsl.exe -d openSUSE-Tumbleweed -u root -e /mnt/wsl/docker-desktop/docker-desktop-user-distro --docker-desktop-root /mnt/wsl/docker-desktop --distro-name openSUSE-Tumbleweed cleanup-cli-tools
> -------------------------------------------------------------------------------->8
> -------------------------------------------------------------------------------->8
> [2023-10-10T00:56:14.922696300Z][com.docker.wsl-distro-proxy.exe][I] kill leaked integration processes for distro Arch
> [2023-10-10T00:56:14.924861800Z][com.docker.wsl-distro-proxy.exe.linuxworkspaces][I] wsl.exe -d Arch -u root -e sh -c ps aux
> -------------------------------------------------------------------------------->8
> -------------------------------------------------------------------------------->8
> [2023-10-10T00:56:15.308897600Z][com.docker.wsl-distro-proxy.exe][I] kill leaked integration processes for distro openSUSE-Tumbleweed
> [2023-10-10T00:56:15.309421800Z][com.docker.wsl-distro-proxy.exe.linuxworkspaces][I] wsl.exe -d openSUSE-Tumbleweed -u root -e sh -c ps aux
> ```

I reinstalled v4.24.2, turned on WSL2 integration for Arch only. It immediately fixed the issue, here's the logs from the `com.docker.wsl-distro-proxy.exe` immediately after the version change.

Something is very wrong with v4.25.0

```
[2023-11-05T18:03:00.105168300Z][com.docker.wsl-distro-proxy.exe][I] start wsl integration for distro Arch
[2023-11-05T18:03:00.105688100Z][com.docker.wsl-distro-proxy.exe.linuxworkspaces][I] wsl.exe -d Arch -e sh -c mkdir -p ~/.docker/run
[2023-11-05T18:03:00.258939200Z][com.docker.wsl-distro-proxy.exe.linuxworkspaces][I] wsl.exe -d Arch -e sh -c cat ~/.docker/config.json
[2023-11-05T18:03:00.397969400Z][com.docker.wsl-distro-proxy.exe.linuxworkspaces][I] wsl.exe -d Arch -e sh -c cat - > ~/.docker/config.json
[2023-11-05T18:03:00.538880500Z][com.docker.wsl-distro-proxy.exe.linuxworkspaces][I] wsl.exe -d Arch -e sh -c mkdir -p ~/.docker/scan
[2023-11-05T18:03:00.680953200Z][com.docker.wsl-distro-proxy.exe.linuxworkspaces][I] wsl.exe -d Arch -e sh -c cat ~/.docker/scan/config.json
[2023-11-05T18:03:00.822574500Z][com.docker.wsl-distro-proxy.exe.linuxworkspaces][I] wsl.exe -d Arch -e sh -c cat - > ~/.docker/scan/config.json
[2023-11-05T18:03:00.980429400Z][com.docker.wsl-distro-proxy.exe.linuxworkspaces][I] wsl.exe -d Arch -e whoami
[2023-11-05T18:03:01.135536300Z][com.docker.wsl-distro-proxy.exe.linuxworkspaces][I] wsl.exe -d Arch -u root -e sh -c ( addgroup docker || true ) && addgroup "zoltanmagyar" docker
[2023-11-05T18:03:01.276146600Z][com.docker.wsl-distro-proxy.exe.linuxworkspaces][I] wsl.exe -d Arch -u root -e sh -c ( groupadd docker || true ) && usermod -aG docker "zoltanmagyar"
[2023-11-05T18:03:01.417628300Z][com.docker.wsl-distro-proxy.exe.linuxworkspaces][I] wsl.exe -d Arch -e cat /etc/wsl.conf
[2023-11-05T18:03:01.555850100Z][com.docker.wsl-distro-proxy.exe.linuxworkspaces][I] wsl.exe -d Arch -e cat /etc/wsl.conf
[2023-11-05T18:03:01.709870000Z][com.docker.wsl-distro-proxy.exe.linuxworkspaces][I] wsl.exe -d Arch -e /mnt/wsl/docker-desktop/docker-desktop-user-distro sync-context-store /mnt/c/Users/zoltanmagyar/.docker/contexts
[2023-11-05T18:03:01.865047400Z][com.docker.wsl-distro-proxy.exe.linuxworkspaces][I] wsl.exe -d Arch -e cat /etc/wsl.conf
[2023-11-05T18:03:01.988354600Z][com.docker.wsl-distro-proxy.exe.linuxworkspaces][I] wsl.exe -d Arch -e cat /etc/wsl.conf
[2023-11-05T18:03:02.112423700Z][com.docker.wsl-distro-proxy.exe.linuxworkspaces][I] wsl.exe -d Arch -e /mnt/wsl/docker-desktop/docker-desktop-user-distro sync-home-subdir /mnt/c/Users/zoltanmagyar/.aws
[2023-11-05T18:03:02.236917300Z][com.docker.wsl-distro-proxy.exe.linuxworkspaces][I] wsl.exe -d Arch -e cat /etc/wsl.conf
[2023-11-05T18:03:02.361609200Z][com.docker.wsl-distro-proxy.exe.linuxworkspaces][I] wsl.exe -d Arch -e cat /etc/wsl.conf
[2023-11-05T18:03:02.484854200Z][com.docker.wsl-distro-proxy.exe.linuxworkspaces][I] wsl.exe -d Arch -e /mnt/wsl/docker-desktop/docker-desktop-user-distro sync-home-subdir /mnt/c/Users/zoltanmagyar/.azure
[2023-11-05T18:03:02.638356200Z][com.docker.wsl-distro-proxy.exe.linuxworkspaces][I] wsl.exe -d Arch -e cat /etc/wsl.conf
[2023-11-05T18:03:02.776070700Z][com.docker.wsl-distro-proxy.exe.linuxworkspaces][I] wsl.exe -d Arch -e cat /etc/wsl.conf
[2023-11-05T18:03:02.915124700Z][com.docker.wsl-distro-proxy.exe.linuxworkspaces][I] wsl.exe -d Arch -e /mnt/wsl/docker-desktop/docker-desktop-user-distro sync-file /mnt/c/Users/zoltanmagyar/.docker/features.json ~/.docker/features.json
[2023-11-05T18:03:03.053095600Z][com.docker.wsl-distro-proxy.exe.linuxworkspaces][I] wsl.exe -d Arch -e sh -c echo $HOME
[2023-11-05T18:03:03.193210900Z][com.docker.wsl-distro-proxy.exe.linuxworkspaces][I] Killing CLI API server running at /home/zoltanmagyar/.docker/run/docker-cli-api.sock
[2023-11-05T18:03:03.193210900Z][com.docker.wsl-distro-proxy.exe.linuxworkspaces][I] wsl.exe -d Arch -e sh -c ps -x
[2023-11-05T18:03:03.331407000Z][com.docker.wsl-distro-proxy.exe.linuxworkspaces][I] wsl.exe -d Arch -e ls /home/zoltanmagyar/.docker/run/docker-cli-api.sock
[2023-11-05T18:03:03.454629700Z][com.docker.wsl-distro-proxy.exe.linuxworkspaces][I] Existing cli api socket file /home/zoltanmagyar/.docker/run/docker-cli-api.sock, trying to clean it
[2023-11-05T18:03:03.454629700Z][com.docker.wsl-distro-proxy.exe.linuxworkspaces][I] wsl.exe -d Arch -e rm /home/zoltanmagyar/.docker/run/docker-cli-api.sock
[2023-11-05T18:03:03.578088900Z][com.docker.wsl-distro-proxy.exe][I] Starting WSL Distro proxy 2
[2023-11-05T18:03:03.578088900Z][com.docker.wsl-distro-proxy.exe.linuxworkspaces][I] wsl.exe -d Arch -e cat /etc/wsl.conf
[2023-11-05T18:03:03.701700800Z][com.docker.wsl-distro-proxy.exe.linuxworkspaces][I] wsl.exe -d Arch -u root -e /mnt/wsl/docker-desktop/docker-desktop-user-distro proxy --distro-name Arch --docker-desktop-root /mnt/wsl/docker-desktop C:\Program Files\Docker\Docker\resources
[2023-11-05T18:03:03.828815800Z][com.docker.wsl-distro-proxy.exe.linuxworkspaces][I] Starting distro cli api server on /home/zoltanmagyar/.docker/run/docker-cli-api.sock
[2023-11-05T18:03:03.828815800Z][com.docker.wsl-distro-proxy.exe.linuxworkspaces][I] wsl.exe -d Arch -e docker serve --address unix:///home/zoltanmagyar/.docker/run/docker-cli-api.sock

```

**Comment by iskorotkov** [comment]
I am using the [build from this thread](https://github.com/docker/for-win/issues/13764#issuecomment-1790616576) with Ubuntu.

Though installing this build didn't do a thing for me, changing login shell from fish to bash solves the issue for me. Maybe docker integration doesn't work correctly with fish?

I'm not sure whether changing login shell will work on 4.25 without modifications as I already installed the provided build prior to going back to bash.

**Comment by chelnak** [comment]
> I am using the [build from this thread](https://github.com/docker/for-win/issues/13764#issuecomment-1790616576) with Ubuntu.
> 
> 
> 
> Though installing this build didn't do a thing for me, changing login shell from fish to bash solves the issue for me. Maybe docker integration doesn't work correctly with fish?
> 
> 
> 
> I'm not sure whether changing login shell will work on 4.25 without modifications as I already installed the provided build prior to going back to bash.

Hey!

Yeah that's right. We should have a test build ready with a fix for you all to try out a bit later today.

**Comment by parberge** [comment]
> @tmuka @agarzon @parberge Thank you!
> 
> Back to the drawing board then.
> 
> @parberge can you send a diagnostic when you get some free time? I'm interested that something seemed to work then stopped.
> 
> Also, if you haven't already, can you all try a factory reset please?

I'm not even sure anymore that I got it working with any 4.25.0 release. I can't afford more downtime so I downgraded. Hopefully you have enough diagnostics from the others that have responded.

**Comment by chelnak** [comment]
@parberge The diagnostics gathered from everyone on this thread have been very useful and much appreciated. 🙂 

I'll post a link to the new build shortly.

**Comment by chelnak** [comment]
👋 Hello,

This build should address the issues with integration services. 🤞 

https://desktop-stage.docker.com/win/main/amd64/127466/Docker%20Desktop%20Installer.exe

**Comment by netrusov** [comment]
@chelnak can confirm that this build solved this issue for me (WSL2, Ubuntu 20.04, fish-shell). thanks 👍

**Comment by agarzon** [comment]
> 👋 Hello,
> 
> This build should address the issues with integration services. 🤞
> 
> https://desktop-stage.docker.com/win/main/amd64/127466/Docker%20Desktop%20Installer.exe

Tested and it works. 🎉

WSL2 Ubuntu 20.04, Windows 10, shell: fish

**Comment by mattgagliardi** [comment]
I was experiencing the same/a similar issue, can confirm that updated build resolved my problem.

**Comment by parberge** [comment]
> 👋 Hello,
> 
> This build should address the issues with integration services. 🤞 
> 
> https://desktop-stage.docker.com/win/main/amd64/127466/Docker%20Desktop%20Installer.exe
> 

Yes worked for me as well 👍

**Comment by carlosrmendes** [comment]
same for me, when my docker desktop v4.25.0 is in resource saver mode I cannot execute the "docker" command on WSL terminal

**Comment by chelnak** [comment]
Hey @carlosrmendes ,

Check out the build here 😀:

https://github.com/docker/for-win/issues/13764#issuecomment-1798242872

**Comment by lorenrh** [comment]
We've released a new version of Docker Desktop containing a fix for this issue, you can see [the release notes](https://docs.docker.com/desktop/release-notes/)! I'll close this issue for now, but if the issue persists, feel free to open a new issue!

**Comment by tthierryEra** [comment]
I updated to the latest docker version 4.25.1 and still got this issue... 

I always have to restart Docker...

**Comment by carlosrmendes** [comment]
same for me, still got the issue @chelnak

**Comment by captainkirkoff** [comment]
Still got the issue in 4.25.1
As a workaround, since it appeared in 4.25, I can only go to Settings->Resources->WSL Integration, disable default and additional integration, Apply & restart, re-enable integration, re-Apply & restart.

**Comment by chelnak** [comment]
Thanks for the updates everyone! Sorry to hear that it's still not working as expected.

Would you be able to send over some diagnostics?

**Comment by Meisamrzp** [comment]
https://github.com/docker/for-win/issues/13764#issuecomment-1793606728

**Comment by Meisamrzp** [comment]
https://github.com/docker/for-win/issues/13764#issuecomment-1793606728

**Comment by rgilliam-nex** [comment]
> Thanks for the updates everyone! Sorry to hear that it's still not working as expected.
> 
> Would you be able to send over some diagnostics?

@chelnak it's exactly the same issue as before

**Comment by Hades32** [comment]
v4.26.1 on Win11 - same experience

**Comment by chelnak** [comment]
@Hades32 can you send us a diagnostic and post the id here?

Also, are you using fish as your shell?

**Comment by cipriantarta** [comment]
v4.26.1 on Win11 - still can't run contianers from inside WSL
I use `fisth`

How do I provide diags?

**Comment by chelnak** [comment]
Hey @cipriantarta 

You can follow the steps here to submit them either via the terminal or via the Docker Desktop UI.

https://docs.docker.com/desktop/troubleshoot/overview/#diagnose

You'll get an ID once the collection is complete and you can post it here.

**Comment by cipriantarta** [comment]
@chelnak , this is my diags ID: 457C995C-4680-4CB4-AAB8-716FA2A90DCE/20240110064047

**Comment by chelnak** [comment]
@cipriantarta Thanks for sending those over.

It's a strange one, the issue does look similar.

My theory is that you have an old integration agent running that for some reason did not get properly updated. I see no obvious indications as to why that would have happened in the logs though.

Would you be able to try the following and let us know the results?

### Disable/Re-Enable WSL integration
1. Disable WSL integration for each of your distros listed in Settings > Resources > WSL integration
2. Quit Docker Desktop
3. Start Docker Desktop. Once started, re enable WSL integration for the relevant distros

If the result is still the same, then try

### Reset to factory defaults
> edit: clean / purge data should now work here!
1. From the troubleshoot screen, Reset to factory defaults: https://docs.docker.com/desktop/troubleshoot/overview/#troubleshoot-menu

Note on the above - You might want to back up your settings.json before you factory reset.

``` PowerShell
Copy-Item -Path $ENV:APPDATA\Docker\settings.json -Destination $HOME\settings.json
```

**Comment by cipriantarta** [comment]
So none of that worked. I even uninstalled docker-desktop completely.
After a fresh install, the WSL integration was enabled by default but running `docker ps` from inside my Ubuntu WSL still gives the same issue.

This is very strange. I might need to upgrade my Ubuntu WSL, but don't have time to deal with that any time soon. I did remember when you said that I have an old agent running, that I have actually exported/imported by WSL image from after a fresh Windows install some very long time ago. Not sure if that's the issue, but it is very strange if it is :)

Everything was running smoothly tho before 4.25.

Edit:
Some additional information.
- I've disabled WSL integration
- In my Ubuntu distro after disabling the integration, doing:
```fish
❯ ls -alh $(which docker)
lrwxrwxrwx 1 root root 48 Jan  8 10:01 /usr/bin/docker -> /mnt/wsl/docker-desktop/cli-tools/usr/bin/docker*
```
- I installed a Debian distro and doing the same `ls`, shows nothing.
- I then enabled WSL integration. It works correctly in the new Debian distro, but same issue as before in my Ubuntu one

I wonder how does the docker binary get injected in my Ubuntu distro if WSL integration is disabled. Thoughts?

So now I have this:
Debian:
```fish
❯ ls -alh $(which docker)
lrwxrwxrwx 1 root root 48 Jan 11 18:57 /usr/bin/docker -> /mnt/wsl/docker-desktop/cli-tools/usr/bin/docker*
```

Ubuntu:
```fish
❯ ls -alh $(which docker)
lrwxrwxrwx 1 root root 48 Jan  8 10:01 /usr/bin/docker -> /mnt/wsl/docker-desktop/cli-tools/usr/bin/docker*
```

Looking at the modified date we have Jan 8 & Jan 11, what's going on?

Edit 2:

I just removed the simlinks from Ubuntu, restarted WSL and now it works

**Comment by xiao-linxin** [comment]
The version is 4.31.1 (153621) 
```
$ docker

The command 'docker' could not be found in this WSL 2 distro.
We recommend to activate the WSL integration in Docker Desktop settings.

For details about using Docker Desktop with WSL 2, visit:

https://docs.docker.com/go/wsl2/

$ ls -alh $(which docker)
lrwxrwxrwx 1 root root 48 Jul  4 16:26 /Docker/host/bin/docker
```
Try a solution

```
$ sudo ln -s /mnt/wsl/docker-desktop/cli-tools/usr/bin/docker /Docker/host/bin/docker
$ docker --version
Docker version 26.1.4, build 5650f9b
```

Just fine

**Comment by learning310** [comment]
I encounter the same situation, i solve it by using wsl builtin kernel instead of the kernel compile by myself for custom feauture. Hope it can help some of you guys.
